adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -345,9 +345,9 @@ String getVersion4Uuid(String fileUrl, Date modificationTime) {
   }
 
   /**
-   * Returns the files meeting the criteria for replication by removing elements that: 1) are of
-   * type DIRECTORY or 2) are not one of the failed IDs and have a modification time before or equal
-   * to the filter date, when the filter date is specified
+   * Returns the files meeting the criteria for replication by only including elements that: 1) are
+   * of type FILE or 2) are one of the failed IDs or 3) have a modification time after the filter
+   * date
```

Comment:
Consider rephrasing this to indicate how the logic evaluation is grouped for better clarity.

`... only including elements that: 1) are of type FILE and 2) are one of the failed IDs or have a modification time after the filter date`